src/dynamite/__init__.py

```diff
@@ -107,9 +107,8 @@ def _initialize(self, slepc_args=None, version_check=True, gpu=None):
             '-options_left', '0'
         ]
 
-        # prevent PETSc from being sad if we don't use gpu aware mpi
         if not self.initialized and bbuild.have_gpu_shell():
-            slepc_args += ['-use_gpu_aware_mpi', '0'] # we only use one process anyway
+            slepc_args += ['-use_gpu_aware_mpi', '1']
```

Comment:
Tentative plan:
 - if we have 1 rank and are using GPU, turn off GPU aware MPI, it doesn't matter
 - if we have multiple ranks, just don't set the flag at all. Hopefully then it should use GPU aware MPI if available, and print a warning otherwise